ci/scripts/integration-test.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+require() {
+	if [ "$1" == "" ]
+	then
+		echo "The environment variable $2 must be present and contain $3"
+		exit 4
+	fi
+}
+
+section() {
+	echo -e "\n============================\n= $1\n============================\n"
+}
+
+SCRIPTDIR=$(dirname $0)
+SCRIPTDIR=$(realpath $SCRIPTDIR)
+
+section "Checking config"
+require "$SERVICE_ACCOUNT" "SERVICE_ACCOUNT" "the service account JSON pubsubbeat will use"
+require "$PROJECT_ID" "PROJECT_ID" "the GCP project ID"
+require "$TOPIC_NAME" "TOPIC_NAME" "the name of the integration test pubsub topic"
+require "$SUBSCRIPTION_NAME" "SUBSCRIPTION_NAME" "the name of the integration test subscription"
+
+section "Building"
+source $SCRIPTDIR/setup-go.sh
+make
+
+section "Pulling down gcloud"
+source $SCRIPTDIR/setup-gcloud.sh
+
+section "Setting up environment"
+export TESTID=`date +%s`
+export TESTDIR="/tmp/$TESTID"
+export KEYPATH="$TESTDIR/key.json"
+
+mkdir -p $TESTDIR
+
+echo "Test ID: $TESTID"
+echo $SERVICE_ACCOUNT > $KEYPATH
+
+section "Clearing any existing messages from $SUBSCRIPTION_NAME"
+gcloud pubsub subscriptions pull --limit 100 --auto-ack "projects/$PROJECT_ID/subscriptions/$SUBSCRIPTION_NAME"
```

Comment:
It appears that if the pubsub subscription doesn't already exist in the given project, this command will fail. If the intention is to provide a zero config integration testing experience, (beyond creating the GCP project), then I'd recommend a pre-task which creates the pubsub subscript and topic, and a post-task for tear down. See: https://cloud.google.com/pubsub/docs/quickstart-cli

Reply:
Bonus points for using Terraform which will gracefully handle the case when the sub/topic already exist ;)